test/Remittance.js

```diff
@@ -0,0 +1,339 @@
+            const txObj = await instance.newTransaction(exchange, hashedOTP, fiveDays, {from: stranger, value: toWei("0.1", "ether")});
+            assert.strictEqual(txObj.logs.length, 2);
+            assert.strictEqual(txObj.logs[0].event, "LogTakeFees");
+            assert.strictEqual(txObj.logs[0].args[0], stranger);
+            assert.strictEqual(txObj.logs[0].args[1].toString(), "2000");
+    
+            assert.strictEqual(txObj.logs[1].event, "LogNewTransaction");
+            assert.strictEqual(txObj.logs[1].args[0], stranger);
+            assert.strictEqual(txObj.logs[1].args[1], exchange);
+            assert.strictEqual(txObj.logs[1].args[2].toString(), new BN(toWei("0.1", "ether")).minus(2000).toString());
+            
+            const order = await instance.transactions.call(hashedOTP);
+            assert.strictEqual(order.amount.toString(), new BN(toWei("0.1", "ether")).minus(2000).toString())
+
+            const fees = await instance.fees.call(owner);
+            assert.strictEqual(fees.toString(), "2000");
+        });
+    
+        it('Should fail if the address is invalid', async () => {
+            await truffleAssert.reverts(
+                instance.newTransaction(nil, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            );       
+        });
+
+        it('Should fail if the hashOTP is invalid', async () => {
+            await truffleAssert.reverts(
+                instance.newTransaction(exchange, web3.utils.utf8ToHex(""), fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            );          
+        });
+
+        it('Should fail if the tx value is equal to 0', async () => {
+            await truffleAssert.reverts(
+                instance.newTransaction(exchange, password, fiveDays, {from: owner})
+            );          
+        });
+
+        it('Should fail if the hashedOTP is already used', async () => {
+            await instance.newTransaction(exchange, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            await truffleAssert.reverts(
+                instance.newTransaction(exchange, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")})
+            );
+        });
+    })
+
+    describe('======= cancelTransaction unit testing =======', () => {
+        let hashedOTP;
+
+        beforeEach('Creating new transaction', async () => {
+            hashedOTP = await instance.hashOTP(exchange, password);
+
+            const txObj = await instance.newTransaction(exchange, hashedOTP, fiveDays, {from: owner, value: toWei("0.1", "ether")});
+
+            assert.strictEqual(txObj.logs.length, 2);
+            assert.strictEqual(txObj.logs[0].event, "LogTakeFees");
+            assert.strictEqual(txObj.logs[0].args[0], owner);
+            assert.strictEqual(txObj.logs[0].args[1].toString(), "2000");
+            assert.strictEqual(txObj.logs[1].event, "LogNewTransaction");
```

Comment:
All this was already tested elsewhere. Better leave the `beforeEach` do straight actions.

Reply:
OK I was not confident about this point. Thanks to cleared this out :)